pkg/controllers/mcs/serviceimport_controller.go

```diff
@@ -0,0 +1,374 @@
+				klog.Errorf("Create delicate service(%s/%s) failed, for: %v", newService.Namespace, newService.Name, err)
+				return err
+			}
+			if err = c.updateServiceStatus(svcImport, derivedService); err != nil {
+				klog.Errorf("Update derived service(%s/%s) status failed, for %v", newService.Namespace, newService.Name, err)
+				return err
+			}
+		}
+		return err
+	}
+
+	if !reflect.DeepEqual(derivedService.Spec.Ports, newService.Spec.Ports) {
+		derivedService.Spec.Ports = newService.Spec.Ports
+		if derivedService, err = c.localk8sClient.CoreV1().Services(svcImport.Namespace).Update(context.TODO(), derivedService, metav1.UpdateOptions{}); err != nil {
+			klog.Errorf("Update derived service(%s/%s) spec failed, for %v", derivedService.Namespace, derivedService.Name, err)
+			return err
+		}
+	}
+
+	if err = c.updateServiceStatus(svcImport, derivedService); err != nil {
+		klog.Errorf("Update derived service(%s/%s) status failed, for %v", newService.Namespace, newService.Name, err)
+		return err
+	}
+
+	return nil
+}
+
+// updateServiceStatus update ServiceStatus with retry.
+func (c *ServiceImportController) updateServiceStatus(svcImport *v1alpha1.ServiceImport, derivedService *corev1.Service) error {
+	klog.V(5).Infof("try to update Service %q status", derivedService.Name)
+	// update loadbalanacer status with provided clusterset IPs
+	var ingress []corev1.LoadBalancerIngress
+	for _, ip := range svcImport.Spec.IPs {
+		ingress = append(ingress, corev1.LoadBalancerIngress{
+			IP: ip,
+		})
+	}
+
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		derivedService.Status = corev1.ServiceStatus{
+			LoadBalancer: corev1.LoadBalancerStatus{
+				Ingress: ingress,
+			},
+		}
+		_, err := c.localk8sClient.CoreV1().Services(derivedService.Namespace).UpdateStatus(context.TODO(), derivedService, metav1.UpdateOptions{})
+		if err == nil {
+			return nil
+		}
+
+		if updated, err := c.localk8sClient.CoreV1().Services(derivedService.Namespace).Get(context.TODO(), derivedService.Name, metav1.GetOptions{}); err == nil {
+			// make a copy so we don't mutate the shared cache
+			derivedService = updated.DeepCopy()
+		} else {
+			utilruntime.HandleError(fmt.Errorf("error getting updated Service %q from lister: %v", derivedService.Name, err))
+		}
+		return err
+	})
+}
+
+func (c *ServiceImportController) Run(ctx context.Context) {
+	c.mcsInformerFactory.Start(ctx.Done())
+	controller := yacht.NewController("serviceimport").
+		WithCacheSynced(c.serviceImportInformer.Informer().HasSynced, c.endpointSliceInformer.Informer().HasSynced).
+		WithHandlerFunc(c.Handle).WithEnqueueFilterFunc(preFilter)
+	c.serviceImportInformer.Informer().AddEventHandler(controller.DefaultResourceEventHandlerFuncs())
+	c.endpointSliceInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: func(obj interface{}) bool {
+			if _, err := c.getServiceImportFromEndpointSlice(obj); err == nil {
```

Comment:
It seems we don't care about the event type on `EndpointSlice`. So shall we just enqueue the object during filterFunc ?